addons/binding/org.openhab.binding.tesla/ESH-INF/thing/models.xml

```diff
@@ -0,0 +1,718 @@
+<!-- 			<channel id="seattype" typeId="seattype" />			
+ -->			<channel id="shiftstate" typeId="shiftstate" />
+			<channel id="smartpreconditioning" typeId="smartpreconditioning" />
+			<channel id="soc" typeId="soc" />
+			<channel id="speed" typeId="speed" />
+			<channel id="state" typeId="state" />
+<!-- 			<channel id="sunroofinstalled" typeId="sunroofinstalled" />
+ -->			<channel id="sunroofstate" typeId="sunroofstate" />
+			<channel id="sunroof" typeId="sunroof" />
+			<channel id="temperature" typeId="temperature" />
+			<channel id="timetofullcharge" typeId="timetofullcharge" />
+			<channel id="eventstamp" typeId="eventstamp" />
+			<channel id="tripcharging" typeId="tripcharging" />
+			<channel id="usablebatterylevel" typeId="usablebatterylevel" />
+			<channel id="userchargeenablerequest" typeId="userchargeenablerequest" />
+<!-- 			<channel id="vin" typeId="vin" />
+ -->			<channel id="valetmode" typeId="valetmode" />
+			<channel id="valetpin" typeId="valetpin" />
+			<channel id="wakeup" typeId="wakeup" />
+<!-- 			<channel id="wheeltype" typeId="wheeltype" />
+ -->		</channels>
+
+		<config-description>
+			<parameter name="username" type="text" required="true">
+				<label>Network Address</label>
+				<description>Username for the Tesla Remote Service, e.g email address</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Refresh interval</label>
+				<description>Password for the Tesla Remote Service</description>
+			</parameter>
+			<parameter name="vin" type="text" required="true">
+				<label>Vehcile Identification Number</label>
+				<description>VIN of the vehicle associated with this Thing</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="autoconditioning">
+		<item-type>Switch</item-type>
+		<label>Auto Conditioning</label>
+		<description>Indicates if the auto conditioning (A/C only) is switched on</description>
+	</channel-type>
+<!--  	<channel-type id="api" advanced="true">
+		<item-type>Number</item-type>
+		<label>API Version</label>
+		<description>API version used by the vehicle</description>
+	</channel-type>	 -->
+	<channel-type id="batterycurrent" advanced="true">
+		<item-type>Number</item-type>
+		<label>Battery Current</label>
+		<description>Current (Ampere) floating into (+) or out (-) of the battery</description>
+	</channel-type>
+	<channel-type id="batteryheater" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Battery Heater</label>
+		<description>Indicates if the battery heater is switched on</description>
+	</channel-type>
+	<channel-type id="batterylevel">
+		<item-type>Number</item-type>
+		<label>Battery Level</label>
+		<description>State of the battery in %</description>
+	</channel-type>
+	<channel-type id="batteryrange" advanced="true">
+		<item-type>Number</item-type>
+		<label>Battery Range</label>
+		<description>Range, in km or miles, of the battery</description>
```

Comment:
The car is reporting back how the odometer is displayed, e.g. miles or km. This is set in the car itself, and I use it in the channel selector to get to the right value. Since the USA is the center of the world, everything is reported as miles or miles/hr by the car.